data-entry/invoice-editor/invoice-editor-flow/src/main/java/com/vaadin/invoice/editor/InvoiceEditor.java

```diff
@@ -152,7 +156,14 @@ public InvoiceEditor() {
         grid.addComponentColumn(item -> createRemoveButton(grid, item))
                 .setWidth("70px").setFlexGrow(0).setTextAlign(ColumnTextAlign.CENTER);
         grid.getColumns().forEach(column -> column.setResizable(true));
-        
+
+        addCardTransactionBtn.addClickListener(event -> {
+        	ListDataProvider<Invoice> dataProvider = (ListDataProvider<Invoice>) grid.getDataProvider();
```

Comment:
The logic of adding new row should be aligned with the frontend example. (Should be added at the beginning)